cmd/sync/main.go

```diff
@@ -0,0 +1,164 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/umayr/sync"
+)
+
+var (
+	flagSubreddit    string
+	flagPages        int
+	flagRedirectURI  string
+	flagPlaylistID   string
+	flagRefreshToken string
+)
+
+func init() {
+	flag.StringVar(
+		&flagPlaylistID,
+		"playlist-id",
+		"",
+		"playlist id where tracks are going to be added",
+	)
+	flag.StringVar(
+		&flagRefreshToken,
+		"refresh-token",
+		"",
+		"refresh token to create a new authentication token",
+	)
+	flag.StringVar(
+		&flagSubreddit,
+		"subreddit",
+		"music",
+		"comma separated subreddit names",
+	)
+	flag.IntVar(
+		&flagPages,
+		"pages",
+		3,
+		"max pages to be parsed",
+	)
+	flag.StringVar(
+		&flagRedirectURI,
+		"redirect-url",
+		"http://localhost:8080/callback",
+		"redirect URI registered in spotify application",
+	)
+
+	flag.Parse()
+}
+
+func exitWithErr(err error) {
+	fmt.Println(err.Error())
+	os.Exit(1)
```

Comment:
Use `https://golang.org/pkg/log/#Fatal`

Reply:
Fatal adds a timestamp with every message. That's why it couldn't be used here.